### PR TITLE
Add Front End

### PR DESCRIPTION
**Pull request by BoxOfCereal**

### CODE REVIEW

File: front-end/src/auth/requiresAuth.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    return Promise.reject(err);
+  }
+);
+
+//higher order comp
+export default function(Component) {
+  return props => {
+    const token = localStorage.getItem("jwt");
+    const notLoggedIn = <div>Please Login To See Crap</div>;
+    return <>{token ? <Component {...props} /> : notLoggedIn}</>;
```
Comment (johnoro) [inline front-end/src/auth/requiresAuth.js:27]: Is the fragment necessary here? I think you can just use the ternary:
```javascript
return token ? <Component {...props} /> : notLoggedIn;
```

File: front-end/src/Register/Register.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            <input
+              name="department"
+              id="department"
+              type="text"
+              value={state.department}
+              onChange={e => {
+                dispatch({ type: "SET_DEPARTMENT", payload: e.target.value });
+              }}
+            />
+          </div>
```
Comment (johnoro) [inline front-end/src/Register/Register.js:94]: These input label divs look a bit repetitive; why not make a helper function for them? For example:
```javascript
const makeInputGroup = (name) => (
  <div>
    <label htmlFor={name}>{name[0].toUpperCase() + name.slice(1)}</label>
    <input
      name={name}
      id={name}
      type='text'
      value={state[name]}
      onChange={e => {
        dispatch({ type: `SET_${name.toUpperCase()}`, payload: e.target.value });
      }}
    />
  </div>
);
```

Comment (johnoro) [reply front-end/src/Register/Register.js:94]: and the usage:
```javascript
// ...
<div>
  {makeInputGroup('username')}
  {makeInputGroup('password')}
  {makeInputGroup('department')}
</div>
// ...
```

File: front-end/src/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+              onChange={e => {
+                setPassword(e.target.value);
+              }}
+            />
+          </div>
+        </div>
+        <div>
+          <button type="submit">Login</button>
+        </div>
+      </form>
```
Comment (johnoro) [inline front-end/src/Login/Login.js:58]: This component looks pretty similar to Register; do you think you could combine the like parts into a single component?

**Review by johnoro — APPROVED** [review]
Looking great!

File: front-end/src/Register/Register.js (hunk 2 above)
Comment (BoxOfCereal) [reply front-end/src/Register/Register.js:94]: Thanks for the idea and example! It looks super clean. I'll try to remember that components are just functions and you can use them to generate other components.

File: front-end/src/Login/Login.js (hunk 3 above)
Comment (BoxOfCereal) [reply front-end/src/Login/Login.js:58]: I always have difficulty separating logic in form components so I end up duping them. Any advice is appreciated!

**Pull request merged**